etsd/keys/views.py

```diff
@@ -44,6 +47,34 @@ def get_context_data(self, *args, **kwargs):
 class PublicKeyDetailView(AdminOrAuthorityQsMixin, DetailView):
     model = models.PublicKey
 
+    def post(self, request, *args, **kwargs):
+        pubk= models.PublicKey.objects.get(pk=self.kwargs.get('pk'))
+
+        if "acceptKey" in request.POST:
+            activekeys = models.PublicKey.objects.filter(authority=pubk.authority, status="ACTIVE")
+            for key in activekeys:
+                key.status = "INACTIVE"
+                key.save()
+            pubk.approved_on = datetime.now()
+            pubk.status = "ACTIVE"
+        if "rejectKey" in request.POST:
+            pubk.status = "REJECTED"
+        
+        email_template = get_template("keys/emails/confirmation.txt")
+        email_ctx = dict(fingerprint=pubk.fingerprint, status = pubk.status,)
+        email_body = email_template.render(email_ctx)
```

Comment:
Try moving these 3 lines to a function `core.utils.send_mail_body(email_template, email_ctx)` and use this instead.